### PR TITLE
TFECO-1485: Allow setting TFE_ADDRESS via chart variable

### PR DESCRIPTION
**Pull request by alexsomesan**
### Description

The operator currently defaults to connecting to the public TFC endpoint (app.terraform.io). This change introduces a Helm chart variable that allows users to define a custom API endpoint, which will allow the operator to work with TFE installations as well. 

It makes use of the fact that `go-tfe` observes the TFE_ADDRESS environment variable as a way to set a custom endpoint.
https://pkg.go.dev/github.com/hashicorp/go-tfe@v1.17.0#section-readme

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-cloud-operator/blob/main/CHANGELOG.md):

```release-note
* allow configuring TFE endpoints via "operator.tfe_address" Helm value
```

### Community Note

* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request.
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### CODE REVIEW

**Comment by alexsomesan** [comment]
Tests seem to be failing in main as well... need to look into it.

**Comment by arybolovlev** [comment]
> Tests seem to be failing in main as well... need to look into it.

I have made some updates in the `AgentPool` controller tests. They should be more stable now. The changes are in the main branch already.

File: charts/terraform-cloud-operator/values.yaml [hunk 1]
```diff
@@ -22,6 +22,9 @@ operator:
   # List of namespaces the controllers should monitor
   watchedNamespaces: []
 
+  # The API URL of a TFE instance
+  tfe_address: ""
```
Comment (arybolovlev) [inline charts/terraform-cloud-operator/values.yaml:26]: Should we use here camelCase notation to have consistency in names?

**Comment by arybolovlev** [comment]
One tiny comment about the naming style.

I think we need to update the [FAQ General Question section](https://github.com/hashicorp/terraform-cloud-operator/blob/main/docs/faq.md#general-questions) with something like "Does the Operator work with Terraform Enterprise?" and mention in the answer, that in this case, the Operator deployment will be able to work only with the API point that is configured via `TFC_ADDRESS`, otherwise, with the default one `https://app.terraform.io`. I would even update the top of the README file and mention that the Operator works with both Cloud and Enterprise versions, but for simplicity everywhere it will be "Terraform Cloud" and for more information on how to switch to the Enterprise version -- please read FAQ. :)

Both are just suggestions.

File: charts/terraform-cloud-operator/values.yaml (hunk 1 above)
Comment (alexsomesan) [reply charts/terraform-cloud-operator/values.yaml:26]: Definitely camel case, this was me copy-pasting blindly.

**Review by arybolovlev — APPROVED** [review]
Nice, Alex! Thanks! 👍🏻🚀

**Pull request merged**